fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -292,3 +301,95 @@ impl IBitcoindRpc for ElectrumClient {
         })
     }
 }
+
+pub struct EsploraClient(esplora_client::AsyncClient);
+
+impl EsploraClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        let builder = esplora_client::Builder::new(url.as_str());
+        let client = builder.build_async()?;
+        Ok(Self(client))
+    }
+}
+
+impl fmt::Debug for EsploraClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("EsploraClient")
+    }
+}
+
+#[async_trait]
+impl IBitcoindRpc for EsploraClient {
+    fn backend_type(&self) -> BitcoinRpcBackendType {
+        BitcoinRpcBackendType::Esplora
+    }
+
+    async fn get_network(&self) -> Result<Network> {
+        let genesis_height: u32 = 0;
+        let genesis_hash = self.0.get_block_hash(genesis_height).await?;
+
+        // TODO: How could we extract the BlockHash to a constant, and use something like the `bitcoin::blockdata::constants::genesis_block` from rust-bitcoin ?
+        let network = match genesis_hash.to_hex().as_str() {
+            "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f" => Network::Bitcoin,
+            "000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943" => Network::Testnet,
+            "00000008819873e925422c1ff0f99f7cc9bbb232af63a077a480a3633bee1ef6" => Network::Signet,
+            hash => {
+                warn!("Unknown genesis hash {hash} - assuming regtest");
```

Comment:
This might be a bad assumption if people start running private signets. Don't have a better suggestion, though.

Reply:
It might not even be a problem, alt-net address prefixes are a mess: 

https://github.com/fedimint/fedimint/blob/2af4e6a02f8ca4fb42165bfed263c681c225d9f4/modules/fedimint-wallet/src/lib.rs#L1638-L1647

See also the [cleaner impl landed in rust-bitcoin by now](https://docs.rs/bitcoin/0.29.2/src/bitcoin/util/address.rs.html#818-830).

Reply:
During development, I was wondering if this method would be needed if the network were an env var but I'm not sure if this is a better alternative and/or has already been a discussion in the past.

Reply:
Great, I'll take a look at the impl for the address and see if I can figure out an alternative based on that, otherwise, I'll just extract to a const for now.